infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -154,11 +198,24 @@ private void processor() {
                                 break;
                             case WRITE:
                                 WriteRequest write = (WriteRequest) currOp.getMsg().getPayload();
-                                streamLog.append(write.getGlobalAddress(), (LogData) write.getData());
+                                Runnable append =
+                                        () -> streamLog.append(write.getGlobalAddress(), (LogData) write.getData());
+                                if (writeRecordTimer.isPresent()) {
+                                    writeRecordTimer.get().record(append);
+                                }
+                                else {
+                                    append.run();
+                                }
                                 break;
                             case RANGE_WRITE:
                                 RangeWriteMsg writeRange = (RangeWriteMsg) currOp.getMsg().getPayload();
-                                streamLog.append(writeRange.getEntries());
+                                Runnable appendMultiple = () -> streamLog.append(writeRange.getEntries());
```

Comment:
Let's get rid of code duplication by method extraction
```
private void yay(Runnable fsyncRunnable, Optional<Timer> fsyncTimer) {
        if (fsyncTimer.isPresent()) {
            fsyncTimer.get().record(fsyncRunnable);
        } else {
            fsyncRunnable.run();
        }
    }
```

Reply:
Sounds good